archive/snappy.ksy

```diff
@@ -0,0 +1,57 @@
+meta:
+  id: snappy
+  title: snappy framing format
+  license: CC0-1.0
+  ks-version: 0.9
+  encoding: utf-8
+  endian: le
+doc-ref: https://github.com/google/snappy/blob/master/framing_format.txt
+doc: |
+  snappy's framing2 format
+  
+  Test files can be created with snzip: <https://github.com/kubo/snzip>
+seq:
+  - id: chunks
+    type: chunk
+    repeat: until
+    repeat-until: _io.eof or not _.is_valid
+    # This is ugly, as it eats one extra byte, so an external
+    # program processing this could should take this into account
+types:
+  chunk:
+    seq:
+      - id: identifier
+        type: u1
+        enum: chunk_types
+      - id: body
+        type: chunk_body
+        if: is_valid
+    instances:
+        # more values are actually allowed but in practice
+        # these aren't encountered
+        is_valid:
+          value: identifier == chunk_types::compressed or
+            identifier == chunk_types::uncompressed or
+            identifier == chunk_types::padding or
+            identifier == chunk_types::stream_identifier
+  chunk_body:
+    seq:
+      - id: len_chunk
+        type: int3
+      - id: data
+        size: len_chunk.value
+  int3:
```

Comment:
`s3` maybe

Reply:
also, have recreated it multiple times in multiple specs. How about modularizing it?

Reply:
Personally I would be fine with that. I think that this has come up a few times already in the past. Adding @generalmimon

Reply:
`b24` should also work, right? That wouldn't require an extra type.

Reply:
I hadn't thought of that. I will try!

Reply:
Only for be.

Reply:
> Only for be.

If you mean that `b24` would only work for _big-endian_ (`be`) order, well, you can use `type: b24le` (or `meta/bit-endian: le` + `type: b24`) and it will work for little-endian order just as you'd expect.

This is exactly why I discourage not using an explicit `meta/bit-endian` setting and leaving it set to the default value (`be` because of backward compatibility) in the [User Guide](https://doc.kaitai.io/user_guide.html#bit-endian). If you do that, it works, but it essentially suggests that the big-endian order is the only parsing direction supported.

Reply:
Thanks for the info, I should give it a try!